File: neetcode_150/containsDuplicate.java
import java.util.HashSet;

//https://leetcode.com/problems/contains-duplicate/description/
public class containsDuplicate {
    public static void main(String[] args) {

    }
    public static boolean containsDuplicate(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for(int i : nums){
            if(set.contains(i)){
                return true;
            }
            set.add(i);
        }
        return false;
    }
}

File: neetcode_150/containsDuplicate.py
from typing import List

#This solution isn't as fast as the other one because the other one uses C compiler when we do set(nums). Since this one uses loops, we have to stick with python interpreter which is slower.
# class Solution:
#     def containsDuplicate(self, nums: List[int]) -> bool:
#         dupes = set()
#         for i in range(len(nums)):
#            if dupes.__contains__(nums[i]):
# Used contains originally as I'm switching from java but should never used dunder methods
#             if nums[i] in dupes:
#                 return True
#             dupes.add(nums[i])
#         return False

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        s = set(nums)
        return len(s)!=len(nums)

